tests/base_test_mixins.py

```diff
@@ -513,3 +690,41 @@ def debug_print_response_messages(self, response):
             print("-----------message end-------------\n")
         except KeyError:
             print("\n-------no message----------")
+
+
+class StaticDockerContainerTestMixin(object):
+    @classmethod
+    def setUpClass(cls):  # noqa
+        if six.PY2:
+            from unittest import SkipTest
+            raise SkipTest('Tests for runpy is configured for PY3 only')
+
+        super(StaticDockerContainerTestMixin, cls).setUpClass()
+        cls.static_container_patch = mock.patch(
+            "course.page.code.USE_STATIC_CONTAINER", True)
+        cls.static_container_patch.start()
```

Comment:
Is there a context manager (i.e. `with xyz:`) version of this?

Reply:
Yes, that's
```python
with mock.patch("course.page.code.USE_STATIC_CONTAINER", True)
    blahblah
```
However, when `setUpClass` is done, it will exit the context manager,  the constant value is reset to it's original value `False`, then all tests will be using `True`, which is not expected.

Reply:
In fact we can also patch at the class level, by a decorator for the (mixin) class, but each time we inheretate from that mixin for a testcase, we need to add the decorator for the subclass.

Reply:
To summarize :-) :

First,  I'm faking the server at the class level (in setUpClass() rather than setUp()), for shorter test time.  Second, `mock.patch("course.page.code.USE_STATIC_CONTAINER", True)` should be used when we are faking the runpy server, there's no need to mock `USE_STATIC_CONTAINER` locally (and it's easy to forget to add the decorator or context manager when codeing).

Reply:
The bad side of this is we have to put altogether all tests which use faked server, excluding possible tests which use read docker container.

Reply:
That's fine. We don't have any 'actual Docker' tests just yet, and I'm OK putting that off until we do. It's more important (and easier to debug, and not a security concern) if we do things in-process.